controller/api/destination/server.go

```diff
@@ -482,3 +488,16 @@ func getPodOpaquePortsAnnotations(pod *corev1.Pod) (map[uint32]struct{}, bool, e
 	}
 	return opaquePorts, true, nil
 }
+
+func getPodSkippedInboundPortsAnnotations(pod *corev1.Pod) (map[uint32]struct{}, error) {
+	annotation, ok := pod.Annotations[labels.ProxyIgnoreInboundPortsAnnotation]
+	if !ok {
+		return nil, nil
+	}
+
+	if annotation == "" {
+		return map[uint32]struct{}{}, nil
+	}
+
+	return util.ParsePorts(annotation)
+}
```

Comment:
`toWeightedAddr` checks if this return value contains the `port` key. The map does not need to be initialized for this check, so it can return `nil` when `annotation == ""` as well:
```suggestion
func getPodSkippedInboundPortsAnnotations(pod *corev1.Pod) (map[uint32]struct{}, error) {
	annotation, ok := pod.Annotations[labels.ProxyIgnoreInboundPortsAnnotation]
	if !ok || annotation == "" {
		return nil, nil
	}
	return util.ParsePorts(annotation)
}
```